### PR TITLE
Reflection.Metadata.Tests: make tests pass when rsa+sha1 signing is not supported.

### PR DESCRIPTION
**Pull request by tmds**
Contributes to https://github.com/dotnet/runtime/issues/65874.

@bartonjs @vcsjones ptal.

### CODE REVIEW

File: src/libraries/System.Reflection.Metadata/tests/TestUtilities/SigningUtilities.cs [hunk 1]
```diff
@@ -11,6 +11,32 @@ namespace System.Reflection.PortableExecutable.Tests
...
+        public static bool SupportsSigning
+        {
+            get
+            {
+                if (_supportsSigning == 0)
+                {
+                    bool supported;
+                    try
+                    {
+                        using var rsa = RSA.Create();
```
Comment (vcsjones) [inline src/libraries/System.Reflection.Metadata/tests/TestUtilities/SigningUtilities.cs:25]:
```suggestion
                        using RSA rsa = RSA.Create();
```

**Comment by tmds** [comment]
I'll rebase when https://github.com/dotnet/runtime/pull/70041 is merged, which will get fix the net48 build fail.

**Comment by bartonjs** [comment]
The detection looks good to me.  I don't know enough about the tests to say if suppression is the right answer, or if they should be looking for a particular exception to happen.

The `Checksum` test is calling into TestChecksumAndAuthenticodeSignature, which then calls into SigningUtilities.CalculateRsaSignature, which does the signing.  So, for that test, suppression looks like the right answer (the test can't do what it's supposed to, so it's not testing the product).  I didn't do a similar analysis of the BasicValidationSigned test to see if the test should be suppressed, or adjusted to look for a particular exception happening in these cases.

**Comment by buyaa-n** [comment]
>  I didn't do a similar analysis of the BasicValidationSigned test to see if the test should be suppressed, or adjusted to look for a particular exception happening in these cases.

`BasicValidationSigned` also looks not expecting an exception as it calls  `WritePEImage` with `privateKeyOpt` set
https://github.com/dotnet/runtime/blob/e2cddabc4cc7b8f646472728ada658c922dfb001/src/libraries/System.Reflection.Metadata/tests/PortableExecutable/PEBuilderTests.cs#L147
which also calls `SigningUtilities.CalculateRsaSignature` when `privateKeyOpt` is set
https://github.com/dotnet/runtime/blob/e2cddabc4cc7b8f646472728ada658c922dfb001/src/libraries/System.Reflection.Metadata/tests/PortableExecutable/PEBuilderTests.cs#L91-L94
Then it verifies the PE with `expectedSignature` bytes
https://github.com/dotnet/runtime/blob/e2cddabc4cc7b8f646472728ada658c922dfb001/src/libraries/System.Reflection.Metadata/tests/PortableExecutable/PEBuilderTests.cs#L157-L167

Is there anything else need for sign off @bartonjs @vcsjones?

**Comment by tmds** [comment]
These test fail for the same reason. `SigningUtilities.CalculateRsaSignature` can't calculate the expected signature.

```
 at Interop.Crypto.RsaSignHash(SafeEvpPKeyHandle pkey, RSASignaturePaddingMode paddingMode, IntPtr digestAlgorithm, ReadOnlySpan`1 hash, Span`1 destination) in /home/tester/runtime/src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EvpPkey.Rsa.cs:line 147
 at System.Security.Cryptography.RSAOpenSsl.TrySignHash(ReadOnlySpan`1 hash, Span`1 destination, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding, Boolean allocateSignature, Int32& bytesWritten, Byte[]& signature) in /home/tester/runtime/src/libraries/Common/src/System/Security/Cryptography/RSAOpenSsl.cs:line 806
 at System.Security.Cryptography.RSAOpenSsl.SignHash(Byte[] hash, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding) in /home/tester/runtime/src/libraries/Common/src/System/Security/Cryptography/RSAOpenSsl.cs:line 736
 at System.Security.Cryptography.RSAWrapper.SignHash(Byte[] hash, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding) in /home/tester/runtime/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/RSAWrapper.cs:line 59
 at System.Reflection.PortableExecutable.Tests.SigningUtilities.CalculateRsaSignature(IEnumerable`1 content, Byte[] privateKey) in /home/tester/runtime/src/libraries/System.Reflection.Metadata/tests/TestUtilities/SigningUtilities.cs:line 21
 at System.Reflection.PortableExecutable.Tests.PEBuilderTests.TestChecksumAndAuthenticodeSignature(Stream peStream, Byte[] privateKeyOpt) in /home/tester/runtime/src/libraries/System.Reflection.Metadata/tests/PortableExecutable/PEBuilderTests.cs:line 823
 at System.Reflection.PortableExecutable.Tests.PEBuilderTests.Checksum() in /home/tester/runtime/src/libraries/System.Reflection.Metadata/tests/PortableExecutable/PEBuilderTests.cs:line 755
 at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
 at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr) in /home/tester/runtime/src/libraries/System.Private.CoreLib/src/System/Reflection/MethodInvoker.cs:line 64
```
```
at Interop.Crypto.RsaSignHash(SafeEvpPKeyHandle pkey, RSASignaturePaddingMode paddingMode, IntPtr digestAlgorithm, ReadOnlySpan`1 hash, Span`1 destination) in /home/tester/runtime/src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EvpPkey.Rsa.cs:line 147
 at System.Security.Cryptography.RSAOpenSsl.TrySignHash(ReadOnlySpan`1 hash, Span`1 destination, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding, Boolean allocateSignature, Int32& bytesWritten, Byte[]& signature) in /home/tester/runtime/src/libraries/Common/src/System/Security/Cryptography/RSAOpenSsl.cs:line 806
 at System.Security.Cryptography.RSAOpenSsl.SignHash(Byte[] hash, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding) in /home/tester/runtime/src/libraries/Common/src/System/Security/Cryptography/RSAOpenSsl.cs:line 736
 at System.Security.Cryptography.RSAWrapper.SignHash(Byte[] hash, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding) in /home/tester/runtime/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/RSAWrapper.cs:line 59
 at System.Reflection.PortableExecutable.Tests.SigningUtilities.CalculateRsaSignature(IEnumerable`1 content, Byte[] privateKey) in /home/tester/runtime/src/libraries/System.Reflection.Metadata/tests/TestUtilities/SigningUtilities.cs:line 21
 at System.Reflection.PortableExecutable.Tests.PEBuilderTests.<>c__DisplayClass4_0.<WritePEImage>b__0(IEnumerable`1 content) in /home/tester/runtime/src/libraries/System.Reflection.Metadata/tests/PortableExecutable/PEBuilderTests.cs:line 93
 at System.Reflection.PortableExecutable.PEBuilder.Sign(BlobBuilder peImage, Blob strongNameSignatureFixup, Func`2 signatureProvider) in /home/tester/runtime/src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEBuilder.cs:line 501
 at System.Reflection.PortableExecutable.ManagedPEBuilder.Sign(BlobBuilder peImage, Func`2 signatureProvider) in /home/tester/runtime/src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/ManagedPEBuilder.cs:line 239
 at System.Reflection.PortableExecutable.Tests.PEBuilderTests.WritePEImage(Stream peStream, MetadataBuilder metadataBuilder, BlobBuilder ilBuilder, MethodDefinitionHandle entryPointHandle, Blob mvidFixup, Byte[] privateKeyOpt, Boolean publicSigned, Machine machine, BlobBuilder mappedFieldData) in /home/tester/runtime/src/libraries/System.Reflection.Metadata/tests/PortableExecutable/PEBuilderTests.cs:line 93
 at System.Reflection.PortableExecutable.Tests.PEBuilderTests.BasicValidationSigned() in /home/tester/runtime/src/libraries/System.Reflection.Metadata/tests/PortableExecutable/PEBuilderTests.cs:line 147
 at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
 at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr) in /home/tester/runtime/src/libraries/System.Private.CoreLib/src/System/Reflection/MethodInvoker.cs:line 64
```

**Comment by tmds** [comment]
> Is there anything else need for sign off @bartonjs @vcsjones?

Based on the discussion, I think the PR is good to merge. @bartonjs can you confirm?

**Review by bartonjs — APPROVED** [review]

**Pull request merged**